drv/sidecar-mainboard-controller/src/tofino2.rs

```diff
@@ -231,3 +232,435 @@ impl Sequencer {
         })
     }
 }
+
+bitfield! {
+    #[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+    #[repr(C)]
+    pub struct DebugPortState(u8);
+    pub send_buffer_empty, set_send_buffer_empty: 0;
+    pub send_buffer_full, _: 1;
+    pub receive_buffer_empty, set_receive_buffer_empty: 2;
+    pub receive_buffer_full, _: 3;
+    pub request_in_progress, set_request_in_progress: 4;
+    pub address_nack_error, set_address_nack_error: 5;
+    pub byte_nack_error, set_byte_nack_error: 6;
+}
+
+#[derive(Copy, Clone, PartialEq, FromPrimitive, AsBytes)]
+#[repr(u8)]
+pub enum DebugRequestOpcode {
+    LocalWrite = 0,
+    LocalRead = 32,
+    DirectWrite = 128,
+    DirectRead = 160,
+    IndirectWrite = 192,
+    IndirectRead = 224,
```

Comment:
Maybe put these in binary to make the pattern more clear?
```suggestion
    LocalWrite = 0b0000_0000,
    LocalRead = 0b0010_0000,
    DirectWrite = 0b1000_0000,
    DirectRead =  0b1010_0000,
    IndirectWrite = 0b1100_0000,
    IndirectRead = 0b1110_0000,
```